runtime/common/src/parachains.rs

```diff
@@ -99,13 +112,65 @@ impl<AccountId, T: Currency<AccountId>> ParachainCurrency<AccountId> for T where
 /// Interface to the persistent (stash) identities of the current validators.
 pub struct ValidatorIdentities<T>(rstd::marker::PhantomData<T>);
 
+#[derive(RuntimeDebug, Encode, Decode)]
+#[derive(Clone, Eq, PartialEq)]
+pub struct DoubleVoteReport {
+	/// Identity of the double-voter.
+	pub identity: ValidatorId,
+	/// First vote of the double-vote.
+	pub first: (Statement, ValidityAttestation),
```

Comment:
I'm not convinced that `Statement, ValidityAttestation` pairs make sense. A `ValidityAttestation, CandidateHash` pair would make sense, or a `Statement, Signature` pair.

Reply:
`Statement, Signature` makes most sense, since `ValidityAttestation` does not allow for `Invalid` attestations.